### PR TITLE
sherpa: add new package

### PR DESCRIPTION
**Pull request by vvolkl**
This PR upstreams a recipe for the sherpa generator initially written by @wdconinc for EIC, and extended with build options used in the GENSER project by me.

### CODE REVIEW

File: var/spack/repos/builtin/packages/sherpa/package.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    def configure_args(self):
+        args = []
+        args.append('--enable-shared')
+        args.append('--enable-binreloc')
+        args.append('--enable-static')
+        args.append('--enable-hepevtsize=200000')
+        args.append('--disable-manual')
+        args.append('--with-sqlite3=' + self.spec['sqlite'].prefix)
+        if self.spec.satisfies('+mpi'):
+            args.append('--enable-mpi')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/sherpa/package.py]: Can we shorten this recipe by using:
```suggestion
        args.extend(self.enable_or_disable('mpi'))
```
for variants that match the option to be enabled / disabled?

Comment (vvolkl) [reply var/spack/repos/builtin/packages/sherpa/package.py]: Yes, that's nicer indeed, thanks!

**Comment by wdconinc** [comment]
Sherpa 2.2.12 is now out. Might make sense to add and test. I may have time to do this tomorrow.

File: var/spack/repos/builtin/packages/sherpa/package.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+    homepage = "https://sherpa-team.gitlab.io"
+    url      = "https://gitlab.com/sherpa-team/sherpa/-/archive/v2.2.11/sherpa-v2.2.11.tar.gz"
+    list_url = "https://gitlab.com/sherpa-team/sherpa/-/tags"
+    git      = "https://gitlab.com/sherpa-team/sherpa.git"
+
+    tags = ['hep', 'eic']
+
+    maintainers = ['wdconinc', 'vvolkl']
+
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/sherpa/package.py]:
```suggestion
    version('2.2.12', sha256='4ba78098e45aaac0bc303d1b5abdc15809f30b407abf9457d99b55e63384c83d')
```

Comment (vvolkl) [reply var/spack/repos/builtin/packages/sherpa/package.py]: Thanks @wdconinc, the new version builds ok for me. I'll try to add some of the examples as tests in a future PR.

File: var/spack/repos/builtin/packages/sherpa/package.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    variant('gzip',       default=False, description='Enable gzip support')
+    variant('pythia',     default=True, description='Enable fragmentation/decay interface to Pythia')
+
+    # Note that the delphes integration seems utterly broken: https://sherpa.hepforge.org/trac/ticket/305
+
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('libtool',  type='build')
+    depends_on('m4',       type='build')
+    depends_on('sqlite')
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/sherpa/package.py:70]:
```suggestion
    depends_on('sqlite')
    depends_on('texinfo')
```
Ref: makeinfo needed at https://gitlab.com/sherpa-team/sherpa/-/blob/master/Manual/Makefile.am#L43.

```
make[2]: Entering directory '/home/wdconinc/.spack/stage/spack-stage-sherpa-2.2.11-et52v63lstkwey2ibxw4s2xn4t6i7jey/spack-src/Manual'
make[2]: warning: -j1 forced in makefile: resetting jobserver mode.
Updating ./version.texi
restore=: && backupdir=".am$$" && \
am__cwd=`pwd` && CDPATH="${ZSH_VERSION+.}:" && cd . && \
rm -rf $backupdir && mkdir $backupdir && \
if (/bin/bash '/home/wdconinc/.spack/stage/spack-stage-sherpa-2.2.11-et52v63lstkwey2ibxw4s2xn4t6i7jey/spack-src/missing' makeinfo --version) >/dev/null 2>&1; then \
  for f in Sherpa.info Sherpa.info-[0-9] Sherpa.info-[0-9][0-9] Sherpa.i[0-9] Sherpa.i[0-9][0-9]; do \
    if test -f $f; then mv $f $backupdir; restore=mv; else :; fi; \
  done; \
else :; fi && \
cd "$am__cwd"; \
if /bin/bash '/home/wdconinc/.spack/stage/spack-stage-sherpa-2.2.11-et52v63lstkwey2ibxw4s2xn4t6i7jey/spack-src/missing' makeinfo -I ..  -I . \
 -o Sherpa.info Sherpa.texi; \
then \
  rc=0; \
  CDPATH="${ZSH_VERSION+.}:" && cd .; \
else \
  rc=$?; \
  CDPATH="${ZSH_VERSION+.}:" && cd . && \
  $restore $backupdir/* `echo "./Sherpa.info" | sed 's|[^/]*$||'`; \
fi; \
rm -rf $backupdir; exit $rc
/home/wdconinc/.spack/stage/spack-stage-sherpa-2.2.11-et52v63lstkwey2ibxw4s2xn4t6i7jey/spack-src/missing: line 81: makeinfo: command not found
WARNING: 'makeinfo' is missing on your system.
         You should only need it if you modified a '.texi' file, or
         any other file indirectly affecting the aspect of the manual.
         You might want to install the Texinfo package:
         <https://www.gnu.org/software/texinfo/>
         The spurious makeinfo call might also be the consequence of
         using a buggy 'make' (AIX, DU, IRIX), in which case you might
         want to install GNU make:
         <https://www.gnu.org/software/make/>
make[2]: *** [Makefile:580: Sherpa.info] Error 127
make[2]: Leaving directory '/home/wdconinc/.spack/stage/spack-stage-sherpa-2.2.11-et52v63lstkwey2ibxw4s2xn4t6i7jey/spack-src/Manual'
make[1]: *** [Makefile:571: all-recursive] Error 1
make[1]: Leaving directory '/home/wdconinc/.spack/stage/spack-stage-sherpa-2.2.11-et52v63lstkwey2ibxw4s2xn4t6i7jey/spack-src'
make: *** [Makefile:497: all] Error 2
```

Comment (wdconinc) [reply var/spack/repos/builtin/packages/sherpa/package.py:70]: Actually, just a `type='build'` dependence.

Comment (vvolkl) [reply var/spack/repos/builtin/packages/sherpa/package.py:70]: I was trying to get around this dependency by adding `--disable-manual` to the configure arguments, but it seems that even though it is referenced, that argument is unfortunately not working.

Comment (alalazo) [reply var/spack/repos/builtin/packages/sherpa/package.py:70]: Let me know when this is ready to be merged

Comment (vvolkl) [reply var/spack/repos/builtin/packages/sherpa/package.py:70]: @alalazo I added the missing dependency and reported the manual issue upstream (https://gitlab.com/sherpa-team/sherpa/-/issues/332), unless Wouter has more comments this is good to go!

Comment (wdconinc) [reply var/spack/repos/builtin/packages/sherpa/package.py:70]: Nothing from my side. Looks good to go.

**Review by alalazo — APPROVED** [review]

**Pull request merged**